core/src/main/scala/org/apache/spark/rpc/RpcAddress.scala

```diff
@@ -23,7 +23,9 @@ import org.apache.spark.util.Utils
 /**
  * Address for an RPC environment, with hostname and port.
  */
-private[spark] case class RpcAddress(host: String, port: Int) {
+private[spark] case class RpcAddress(_host: String, port: Int) {
+
+  val host: String = Utils.addBracketsIfNeeded(_host)
```

Comment:
This can be a `lazy val` instead of `val` ?
We end up almost doubling the size of the serialized instance of this object otherwise.

Reply:
Ya, of course, we can. 

Actually, at the first commit, I used `def host` first to save the size . Later I switched to `val` in the second commit.
- https://github.com/apache/spark/pull/36868/commits/ed10d51fc61612d3f7e8441a0219a62b3008e8c7

If the memory consumption of `RpcAddress` instances matter, we can revert back `def host` instead of `lazy val`.

Among three options, which one do you prefer, @mridulm and @cloud-fan ?
1. `def host` (Initial commit)
2. `lazy val host` (New alternative)
3. `val host` (AS-IS)

Reply:
I prefer `lazy val` - it keeps the serialized size the same, while making the runtime cost of accessing `host` is almost the same as before (which is increased by `def host`).

Reply:
Thank you, @mridulm . Here is the followup.
- https://github.com/apache/spark/pull/36882/files